lib/console_web/controllers/router/device_controller.ex

```diff
@@ -139,6 +139,13 @@ defmodule ConsoleWeb.Router.DeviceController do
         result =
           Ecto.Multi.new()
           |> Ecto.Multi.run(:event, fn _repo, _ ->
+            event = case event["data"]["req"]["body"] do
+              nil -> event
+              _ -> 
+                {:ok, decoded_body} = Poison.decode(event["data"]["req"]["body"])
+                Kernel.put_in(event["data"]["req"]["body"], decoded_body)
```

Comment:
can we use Map.put instead?

Reply:
can't do deeply nested using `Map.put` :/

Reply:
ic ic